crates/ide-assists/src/handlers/unwrap_return_type.rs

```diff
@@ -65,11 +55,12 @@ fn unwrap_return_type(
     let Some(hir::Adt::Enum(ret_enum)) = ctx.sema.resolve_type(type_ref)?.as_adt() else {
         return None;
     };
-    let core_enum =
-        kind.core_type(FamousDefs(&ctx.sema, ctx.sema.scope(type_ref.syntax())?.krate()))?;
-    if ret_enum != core_enum {
-        return None;
-    }
+
+    let famous_defs = FamousDefs(&ctx.sema, ctx.sema.scope(type_ref.syntax())?.krate());
+
+    let kind = UnwrapperKind::ALL
+        .iter()
+        .find(|k| matches!(k.core_type(&famous_defs), Some(core_type) if ret_enum == core_type))?;
```

Comment:
If this seems a bit too dense I could easily rewrite this as an `if` that checks against each `UnwrapperKind` individually, just let me know.